mobileapps/models.py

```diff
@@ -0,0 +1,113 @@
+"""
+Django database models supporting the mobile apps
+"""
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+
+from edx_solutions_organizations.models import Organization
+from model_utils.models import TimeStampedModel
+from model_utils.fields import AutoCreatedField
+from edx_solutions_api_integration.utils import StringCipher
+
+
+OS_CHOICES = (
+    (1, 'Android'),
+    (2, 'iOS'),
+    (3, 'Windows'),
+    (4, 'Other'),
+)
+
+DEPLOYMENT_CHOICES = (
+    (1, 'Public app store'),
+    (2, 'Enterprise'),
+    (3, 'OTA'),
+    (4, 'Other'),
+)
+
+
+class NotificationProvider(TimeStampedModel):
+    """
+    A django model to track notification providers.
+    """
+    name = models.CharField(max_length=255)
+    api_url = models.CharField(max_length=255, null=True, blank=True)
+
+
+class MobileApp(TimeStampedModel):
+    """
+    A django model to track mobile apps.
+    """
+    identifier = models.CharField(max_length=255, db_index=True, unique=True)
+    name = models.CharField(max_length=255)
+    operating_system = models.PositiveSmallIntegerField(choices=OS_CHOICES)
+    deployment_mechanism = models.PositiveSmallIntegerField(choices=DEPLOYMENT_CHOICES, default=1)
+    download_url = models.CharField(max_length=255, null=True, blank=True)
+    analytics_url_dev = models.CharField(max_length=255, null=True, blank=True)
+    analytics_url_prod = models.CharField(max_length=255, null=True, blank=True)
+    notification_provider = models.ForeignKey(NotificationProvider, related_name="mobile_apps", blank=True, null=True)
+    provider_key = models.CharField(max_length=255, null=True, blank=True)
+    provider_secret = models.CharField(max_length=255, null=True, blank=True)
+    provider_dashboard_url = models.CharField(max_length=255, null=True, blank=True)
+    current_version = models.CharField(max_length=255)
+    is_active = models.BooleanField(default=True)
+    updated_by = models.ForeignKey(User, on_delete=models.PROTECT, blank=True)
+    users = models.ManyToManyField(User, related_name="mobile_apps", blank=True)
+    organizations = models.ManyToManyField(Organization, related_name="mobile_apps", blank=True)
+
+    @property
+    def get_provider_key(self):
```

Comment:
`get` prefix should be for methods. Can we rename this to  `provider_key_decrypted`? same for `provider_secret`.